### PR TITLE
[RFC] agnostic address proposal

### LINKED ISSUE #197: I want to see the details of historical order So I can deeply analyze it against the one that I received in real life
**Context**
In https://github.com/DivanteLtd/next/issues/190 task we prepared the grid with order history. Now we want to let the user be able to open a single specific order to see more details of it.

**Acceptance Criteria**
* In My Orders (grid view) I can see the possibility to open details of single specific order.
* When I click on the button then I am redirected to the page with order details of single order.
* I can see its billing information.
* I can see its shipping information.
* I can see its status.
* I can see its number.
* I can see order positions with names, quantities, prices, values, discounts.
* I can see summary of the order values (net, vat, gross).

### PR DESCRIPTION
**Pull request by andrzejewsky**
- [x] I'm confirming that if i made any changes to public APIs or exposed any public APIs they are doccumented.

### CODE REVIEW

File: packages/core/rfcs/agnostic-address.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Sometimes we  need to have access to the address property, such as street name, postcode, country etc.
+That is very useful for instance if you want to format these data and display in  the way you want.
+
+```ts
+interface AgnosticAddress {
+  streetName?: string;
+  postcode?: string;
+  city?: string:
+  streetNumber?: string;
+  firstName?: string;
```
Comment (defudef) [inline packages/core/rfcs/agnostic-address.md:14]: what about another field: `apartmentNumber`

File: packages/core/rfcs/agnostic-address.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  phone?: string;
+  company?: string;
+  region?: string;
+  title?: string;
+}
+```
+
+## Migration process
+
+Prety straightforward, there is no address used besides comemrcools orders.
```
Comment (patzick) [inline packages/core/rfcs/agnostic-address.md:25]: in my opinion, the address isn't agnostic, like Product, every platform may have different implementations, shopware example

```ts
export interface BillingAddress {
  countryId: string;
  salutationId: string;
  street: string;
  zipcode: string;
  city: string;
  title?: string;
  additionalAddressLine1?: string;
  additionalAddressLine2?: string;
  phoneNumber?: string;
}

export interface ShippingAddress extends BillingAddress {
  firstName: string;
  lastName: string;
}
```

Comment (andrzejewsky) [reply packages/core/rfcs/agnostic-address.md:25]: still, you can map your fields  into the AgnosticAddress

countryId -> country,
salutationId -> region,
street -> street,
zipcode -> postcode,
city -> city,
title -> title,
additionalAddressLine1, additionalAddressLine2 -> additionalAddres (we can add it)
phoneNumber  -> number

**Comment by rmakara** [comment]
@andrzejewsky @patzick @defudef 
https://youtu.be/DELWr2Cw30g?t=507

TL;DR; problem solved all around the world by postal offices :p do not reinvent the wheel. ;p
